### PR TITLE
(WIP) (RE-5685) Use light.exe to generate MSI

### PR DESCRIPTION
**Pull request by jcoconnor**
Initial PR to create the MSI
Needs #280 Merged first

### CODE REVIEW

File: lib/vanagon/platform/windows.rb [hunk 1]
```diff
@@ -107,7 +107,11 @@ def generate_msi_package(project)
         target_dir = project.repo ? output_dir(project.repo) : output_dir
         cg_name = "ProductComponentGroup"
         dir_ref = "INSTALLDIR"
-        candle_flags =  "-dPlatform=#{@architecture} -arch #{@architecture} -ext WiXUtilExtension -ext WixUIExtension"
+        wix_extensions =  "-ext WiXUtilExtension -ext WixUIExtension"
+        candle_flags =  "-dPlatform=#{@architecture} -arch #{@architecture} #{wix_extensions}"
```
Comment (melissa) [inline lib/vanagon/platform/windows.rb]: Did we decide we want to keep `-dPlatform=#{@architecture}` and use it? Or would it make more sense to always take advantage of erb templates and architecture set there and remove this?

Comment (melissa) [inline lib/vanagon/platform/windows.rb]: I'm partial to removing this so people don't get confused when they're creating their project wix files as to where this comes from/gets set

Comment (jcoconnor) [inline lib/vanagon/platform/windows.rb]: Will put this up as a separate ticket [(RE-6525)](https://tickets.puppetlabs.com/browse/RE-6525) as I need to investigate if there are any other implications, but it looks like a good idea.

**Comment by jcoconnor** [comment]
Updated comments for "flag maintenance" and rebased.

@melissa @McdonaldSeanp Can you review asap so we can merge this up to allow us to light up the build /cc @ScottGarman

**Comment by McdonaldSeanp** [comment]
:+1: pending travis

File: resources/windows/wix/project.wxs.erb [hunk 2]
```diff
@@ -26,9 +26,6 @@
       <ComponentGroupRef Id="ProductComponentGroup" />
     </Feature>
 
-    <!-- All heat runs and WiX Fragment files containing Component elements will reference this ComponentGroup -->
-    <ComponentGroup Id="ProductComponentGroup" Directory="INSTALLDIR"/>
```
Comment (melissa) [inline resources/windows/wix/project.wxs.erb]: Is there a reason this is getting removed? I don't see any explanation in the commits

File: lib/vanagon/platform/windows.rb (hunk 1 above)
Comment (melissa) [inline lib/vanagon/platform/windows.rb]: cool

**Comment by melissa** [comment]
:+1: pending my question about the line removal in `resources/windows/wix/project.wxs.erb`

File: resources/windows/wix/project.wxs.erb (hunk 2 above)
Comment (jcoconnor) [inline resources/windows/wix/project.wxs.erb]: Shouldn't be removed - will fix.

**Comment by ferventcoder** [comment]
this is a test @here

**Pull request merged**